src/main/c/org_apache_activemq_artemis_nativo_jlibaio_LibaioContext.c

```diff
@@ -129,9 +129,24 @@ static int user_io_getevents(io_context_t aio_ctx, unsigned int max,
                     struct io_event *events)
 {
    long i = 0;
-   unsigned head;
+   unsigned head, tail, avail;
    struct aio_ring *ring = to_aio_ring(aio_ctx);
 
+   // continue until the available values are <= max
+   do {
+       head = ring->head;
+       mem_barrier();
+       tail = ring->tail;
+
+       if (tail >= head)
+           avail = tail - head;
+       else
+           avail = ring->nr - (head - tail);
+
+       if (avail <= max)
+           break;
+    } while (1);
```

Comment:
Nothing should be changing the head pointer, so all you need to do is check for updates to tail.  I think what I sent you was about as simple as you can make that loop.  I also think this code change deserves a comment explaining the kernel bug that it is working around.

Reply:
@JeffMoyer I will try to place that loop back. I had an issue on the testsuite and I thought it could be related to this.

Reply:
mmmm you're right, effectively this version won't show that there is just a single writer for `head` and deserve some more comments too, my bad 
@clebertsuconic Please bud, we can use the original version proposed by @JeffMoyer

@JeffMoyer OT: do you think it would be better to replace the reaper loop using `%` with a `memcpy`? If yes I would send a different PR to address it too

Reply:
@clebertsuconic @JeffMoyer I still think it worths to check that tail is not just changed by is "right" too ie < max expected size

Reply:
> @JeffMoyer OT: do you think it would be better to replace the reaper loop using `%` with a `memcpy`? If yes I would send a different PR to address it too

I don't know what you mean.  Maybe just post a diff of what you're proposing?